WcaOnRails/app/models/competition_event.rb

```diff
@@ -21,4 +21,31 @@ def to_wcif
       "rounds" => self.rounds.map(&:to_wcif),
     }
   end
+
+  def load_wcif!(wcif)
+    # Remove extra rounds.
```

Comment:
Yeah, React handles the complexity of diffing and updating for you, which is very nice.
I've attempted to change the code as you suggest, let me know what you think!